cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -491,6 +499,100 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
             return withVoid(CompositeFuture.join(routeFutures));
         }
 
+        Future<ReconciliationState> kafkaExternalBootstrapServiceReady() {
+            if (!kafkaCluster.isExposedWithLoadBalancer() && !kafkaCluster.isExposedWithNodePort()) {
+                return withVoid(Future.succeededFuture());
+            }
+
+            String serviceName = KafkaCluster.externalBootstrapServiceName(name);
+            Future future = Future.future();
+            Future<Void> address = null;
+
+            if (kafkaCluster.isExposedWithNodePort()) {
+                address = serviceOperations.hasNodePort(namespace, serviceName, 1_000, operationTimeoutMs);
+            } else  {
+                address = serviceOperations.hasIngressAddress(namespace, serviceName, 1_000, operationTimeoutMs);
+            }
+
+            address.setHandler(res -> {
+                if (res.succeeded())    {
+                    if (kafkaCluster.isExposedWithLoadBalancer()) {
+                        if (serviceOperations.get(namespace, serviceName).getStatus().getLoadBalancer().getIngress().get(0).getHostname() != null) {
+                            this.kafkaExternalBootstrapAddress = serviceOperations.get(namespace, serviceName).getStatus().getLoadBalancer().getIngress().get(0).getHostname();
```

Comment:
A synchronous `get()`: I wonder what thread executes the handler.

Reply:
I though the whole `createOrUpdate` block is wrapped in `executeBlocking`. But it doesn't seem so. I fixed it.